### PR TITLE
loosen typing_extensions dependency version and update packages

### PR DESCRIPTION
**Pull request by jakul**
Signed-off-by: Craig Blaszczyk <masterjakul@gmail.com>

Solves https://github.com/authzed/authzed-py/issues/37

### CODE REVIEW

**Comment by jakul** [comment]
I have read the CLA Document and I hereby sign the CLA

File: pyproject.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ mock = "^4.0.3"
 protobuf = "^3.19.3"
 protoc-gen-validate = "^0.4.1"
 python = "^3.7"
-typing-extensions = "^3.7.4"
+typing-extensions = ">3.7.4,<5"
```
Comment (samkim) [inline pyproject.toml:22]:
```suggestion
typing-extensions = ">=3.7.4,<5"
```
Just in case there are users depending on 3.7.4

**Review by samkim — COMMENTED** [review]
Tests pass using 4.1.1. Just a small suggestion and should be good to go.

File: pyproject.toml (hunk 1 above)
Comment (jakul) [reply pyproject.toml]: Oops, thanks! I've pushed a new commit with that fixe this

**Review by samkim — APPROVED** [review]
LGTM

**Pull request merged**